packages/core/event.js

```diff
@@ -21,7 +22,7 @@ class BugsnagEvent {
     this.errorClass = stringOrFallback(errorClass, '[no error class]')
     this.errorMessage = stringOrFallback(errorMessage, '[no error message]')
     this.groupingHash = undefined
-    this.metaData = {}
+    this._metadata = {}
```

Comment:
What are the benefits of this mixin pattern compared with pure functions?

For example:

```
constructor ()
  ...
  this._metadata = {}
}

addMetadata (section, ...args) {
    this._metadata = addMetadata(this._metadata, section, ...args);
    return this;
}

```

It's not clear from looking at this class why `this._metadata = {}` is being assigned as it is never mutated in this class but in fact it's crucial for the `'./lib/metadata-delegate'` mixin to work properly, for omitting it would cause a TypeError.

In its current form `metadataDelegate` has a fairly complex API in that requires `this._metadata = {}` defining in the consuming class and calls to its functions being bound so the same context.

Reply:
I think it would be a lie to say it's "pure" but I have pushed an update using your approach, as I think it's much better. We can see if the size diff is ±93 bytes!

Reply:
The diff is now 67 bytes (which includes the removal of all the `return this`) so it's smaller and better 🙌